nvme.c

```diff
@@ -387,6 +387,42 @@ static int check_arg_dev(int argc, char **argv)
 	return 0;
 }
 
+static int open_dev_xnvme(struct nvme_dev **devp, char *devname)
+{
+        int err;
+	struct nvme_dev *dev;
+        struct xnvme_dev *xnvme_handle = NULL;
+	struct xnvme_opts opts = {0};
+	opts.rdwr = 1;
+	opts.create_mode = S_IRUSR | S_IWUSR;
+	opts.nsid = 0x1;
+
+	xnvme_handle = xnvme_dev_open(devname, &opts);
+        if (!xnvme_handle) {
+		printf("xnvme_dev failed \n");
+                err = -1;
+                goto perror;
+        }
+
+	dev = calloc(1, sizeof(*dev));
+        if (!dev)
+                return -1;
+
+	dev->type = NVME_DEV_XNVME;
+	dev->name = basename(devname);
+	dev->direct.hnd.xdev = xnvme_handle;
+	dev->direct.hnd.dev_type = NVME_DEV_XNVME;
+
+	*devp = dev;
+	return 0;
+
+perror:
+        perror(devname);
+        return err;
+}
+
+
+
 static int get_dev(struct nvme_dev **dev, int argc, char **argv, int flags)
```

Comment:
I thought we are going to use the newly introduce transport abstraction from the library through out nvme-cli? Ideally we have this open device in libnvme, no?